sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/metadata/ShardingMetaDataLoader.java

```diff
@@ -103,17 +108,43 @@ private SchemaMetaData loadDefaultSchemaMetaData() throws SQLException {
         return actualDefaultDataSourceName.isPresent()
                 ? SchemaMetaDataLoader.load(dataSourceMap.get(actualDefaultDataSourceName.get()), maxConnectionsSizePerQuery) : new SchemaMetaData(Collections.emptyMap());
     }
-    
-    // TODO check all meta data in once
+
     private void checkUniformed(final String logicTableName, final Map<String, TableMetaData> actualTableMetaDataMap) {
         ShardingTableMetaDataDecorator decorator = new ShardingTableMetaDataDecorator();
         TableMetaData sample = decorator.decorate(actualTableMetaDataMap.values().iterator().next(), logicTableName, shardingRule);
+        Collection<TableMetaDataViolation> metaDataViolations = new LinkedList<>();
```

Comment:
Collection should be LinkedList.

Reply:
Be avoid using get(i) form to iterate operation affect performance, we suggest defining collection type still.